px4_ros2_cpp/src/navigation/experimental/local_navigation_interface.cpp

```diff
@@ -0,0 +1,244 @@
+  aux_local_position.velocity[2] = local_position_estimate.velocity_z.value_or(NAN);
+
+  const Vector2f velocity_xy_variance = local_position_estimate.velocity_xy_variance.value_or(
+    Vector2f{NAN, NAN});
+  aux_local_position.velocity_variance[0] = velocity_xy_variance[0];
+  aux_local_position.velocity_variance[1] = velocity_xy_variance[1];
+  aux_local_position.velocity_variance[2] =
+    local_position_estimate.velocity_z_variance.value_or(NAN);
+
+  // Angular velocity (unused at the moment)
+  aux_local_position.angular_velocity = {NAN, NAN, NAN};
+
+  // Publish
+  aux_local_position.timestamp = _node.get_clock()->now().nanoseconds() * 1e-3;
+  _aux_local_position_pub->publish(aux_local_position);
+}
+
+bool LocalNavigationInterface::isEstimateNonEmpty(const LocalPositionEstimate & estimate) const
+{
+  return estimate.position_xy.has_value() ||
+         estimate.position_z.has_value() ||
+         estimate.velocity_xy.has_value() ||
+         estimate.velocity_z.has_value() ||
+         estimate.attitude_quaternion.has_value();
+}
+
+bool LocalNavigationInterface::isVarianceValid(const LocalPositionEstimate & estimate) const
+{
+  if (estimate.position_xy.has_value() &&
+    (!estimate.position_xy_variance.has_value() ||
+    (estimate.position_xy_variance.value().array() <= 0).any()))
+  {
+    RCLCPP_ERROR(_node.get_logger(), "Estimate value position_xy has an invalid variance value.");
+    return false;
+  }
+
+  if (estimate.position_z.has_value() &&
+    (!estimate.position_z_variance.has_value() || estimate.position_z_variance <= 0))
+  {
+    RCLCPP_ERROR(_node.get_logger(), "Estimate value position_z has an invalid variance value.");
+    return false;
+  }
+
+  if (estimate.velocity_xy.has_value() &&
+    (!estimate.velocity_xy_variance.has_value() ||
+    (estimate.velocity_xy_variance.value().array() <= 0).any()))
+  {
+    RCLCPP_ERROR(_node.get_logger(), "Estimate value velocity_xy has an invalid variance value.");
+    return false;
+  }
+
+  if (estimate.velocity_z.has_value() &&
+    (!estimate.velocity_z_variance.has_value() || estimate.velocity_z_variance <= 0))
+  {
+    RCLCPP_ERROR(_node.get_logger(), "Estimate value velocity_z has an invalid variance value.");
+    return false;
+  }
+
+  if (estimate.attitude_quaternion.has_value() &&
+    (!estimate.attitude_variance.has_value() ||
+    (estimate.attitude_variance.value().array() <= 0).any()))
+  {
+    RCLCPP_ERROR(
+      _node.get_logger(),
+      "Estimate value attitude_quaternion has an invalid variance value.");
+    return false;
+  }
+
+  return true;
+}
+
+bool LocalNavigationInterface::isFrameValid(const LocalPositionEstimate & estimate) const
+{
+  if ((estimate.position_xy.has_value() || estimate.position_z.has_value()) &&
+    _pose_frame == AuxLocalPosition::POSE_FRAME_UNKNOWN)
+  {
+    RCLCPP_ERROR(
+      _node.get_logger(),
+      "Position estimate update requested but pose reference frame is set to POSE_FRAME_UNKNOWN.");
```

Comment:
I would try abbreviate this. Also, this function can in theory be called when you're not requesting an update.
Same applies to velocity below, and the error string in the invalid argument exception ([here](https://github.com/Auterion/px4-ros2-interface-lib/pull/5/files#diff-19dae766cc08d36ce42531a0f6f4460ad4736ada1694ddfd54eb1c9d4e04deccR40)).

```suggestion
      "Position estimate has unknown pose frame.");
```